internal/npm_install/generate_build_file.js

```diff
@@ -1046,19 +1050,32 @@ function printIndexBzl(pkg) {
   let result = '';
   const executables = _findExecutables(pkg);
   if (executables.size) {
-    result = `load("@build_bazel_rules_nodejs//:defs.bzl", "npm_package_bin")
+    result = `load("@build_bazel_rules_nodejs//:defs.bzl", "nodejs_binary", "npm_package_bin")
 
 `;
-  }
-  for (const name of executables.keys()) {
-    result = `${result}
+    const data = [`@${WORKSPACE}//${pkg._dir}:${pkg._name}`];
+    if (pkg._dynamicDependencies) {
+      data.push(...pkg._dynamicDependencies);
+    }
+
+    for (const [name, path] of executables.entries()) {
+      result = `${result}
 
 # Generated helper macro to call ${name}
-def ${name}(**kwargs):
-    npm_package_bin(tool = "@${WORKSPACE}//${pkg._dir}/bin:${name}", **kwargs)
-`;
+def ${name.replace(/-/g, '_')}(**kwargs):
+    if "outs" in kwargs:
+        npm_package_bin(tool = "@${WORKSPACE}//${pkg._dir}/bin:${name}", **kwargs)
```

Comment:
a comment here perhaps?. why `outs` is the determining attribute?

also, do additionalAttributes belong in npm_package_bin? if there is a `"configuration_env_vars"` additional that would likely be needed in both.

Reply:
yes, additionalAttributes are here,line 1073

Reply:
the doc for this has to go somewhere new, since it's a generated macro. I'll make a new doc for all the stuff we generate into your `@npm` and how to use those things